weave/trace_server/table_query_builder.py

```diff
@@ -0,0 +1,103 @@
+from typing import Optional
+
+from weave.trace_server.orm import ParamBuilder
+
+# Constants for table and column names
+TABLE_ROWS_ALIAS = "tr"
+VAL_DUMP_COLUMN_NAME = "val_dump"
+ROW_ORDER_COLUMN_NAME = "row_order"
+
+
+def make_natural_sort_table_query(
```

Comment:
I understand the utility of this optimization, but shouldn't the query engine find the fastest way to get data for you given your params?  There don't seem to be any obvious optimization fences in the other query.

Reply:
i'm not sure i understand this comment. The performance improvement is quite dramatic by applying the limit/offset in the inner query as the free-form val_dumps field is skipped

Reply:
ok, I don't know enough about clickhouse to comment deeply here.

I was just surprised that the query engine doesn't figure out the most performant way for you.